### PR TITLE
feat: #7: support for the insertion of a video bumper

### LINKED ISSUE #7: Insert video bumper
It should be possible to insert a video bumper that is placed first in the resulting VOD. This video bumper is another HLS that is spliced in the same way as the ads but the difference is that it should not be signaled as an ad.

### PR DESCRIPTION
**Pull request by birme**
This PR intends to solve feature #7 where a video bumper can be inserted. A video bumper is not flagged as an ad break and is always first in the resulting VOD.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -144,6 +149,31 @@ class HLSSpliceVod {
...
+    return new Promise((resolve, reject) => {
+      this._parseAdMasterManifest(bumperMasterManifestUri, _injectBumperMasterManifest, _injectBumperMediaManifest)
+      .then(bumper => {
+        const bandwidths = Object.keys(this.playlists);
+        for (let b = 0; b < bandwidths.length; b++) {
+          const bw = bandwidths[b];
+
+          const bumperPlaylist = bumper.playlist[findNearestBw(bw, Object.keys(bumper.playlist))];
+          const bumperLength = bumperPlaylist.items.PlaylistItem.length;
+          let i = 0;
```
Comment (oscnord) [inline index.js]: Does `i` get a value different from 0 anywhere?

File: index.js [hunk 2]
```diff
@@ -144,6 +149,31 @@ class HLSSpliceVod {
...
+          this.bumperDuration = 0;
+          for (let j = 0; j < bumperLength; j++) {
+            this.playlists[bw].items.PlaylistItem.splice(i + j, 0, bumperPlaylist.items.PlaylistItem[j]);
+            this.bumperDuration += (bumperPlaylist.items.PlaylistItem[j].get('duration') * 1000);
+          }
+          this.playlists[bw].items.PlaylistItem[i + bumperLength].set('discontinuity', true);
+          this.playlists[bw].set('targetDuration', this.targetDuration);
+          this.bumperOffset = bumperLength;
```
Comment (oscnord) [inline index.js]: What is `this.bumperOffset` used for?

File: index.js (hunk 1 above)
Comment (birme) [reply index.js]: No actually not as a bumper should always be placed first. I kept it for clarity as it is the same case as for inserting an ad with offset = 0 which would yield an i = 0. Maybe it is more logical to change this...

File: index.js (hunk 2 above)
Comment (birme) [reply index.js]: It is to handle the case where you insert a bumper first and then insert an ad with offset = 0 it should always place it after the video bumper no matter what. So this value stores the offset that the bumper causes all remaining ad inserts.

File: index.js (hunk 1 above)
Comment (birme) [reply index.js]: I have changed this now

File: index.js (hunk 2 above)
Comment (Nfrederiksen) [reply index.js]: So is it used in insertAdAt( )?
how is this different from `this.bumperDuration`  :O ?
I understand that bumperDuration is in ms and bumperOffset is in number of segments.

**Comment by oscnord** [comment]
LGTM

File: index.js [hunk 3]
```diff
@@ -144,6 +149,30 @@ class HLSSpliceVod {
...
+          const bumperPlaylist = bumper.playlist[findNearestBw(bw, Object.keys(bumper.playlist))];
+          const bumperLength = bumperPlaylist.items.PlaylistItem.length;
+          this.bumperDuration = 0;
+          for (let j = 0; j < bumperLength; j++) {
+            this.playlists[bw].items.PlaylistItem.splice(j, 0, bumperPlaylist.items.PlaylistItem[j]);
+            this.bumperDuration += (bumperPlaylist.items.PlaylistItem[j].get('duration') * 1000);
+          }
+          this.playlists[bw].items.PlaylistItem[bumperLength].set('discontinuity', true);
+          this.playlists[bw].set('targetDuration', this.targetDuration);
+          this.bumperOffset = bumperLength;
```
Comment (Nfrederiksen) [inline index.js]: It all looks good and seems to work perfectly. But bumperOffset doesn't look like it's ever used in the insertAdAt( ) function.

**Review by Nfrederiksen — APPROVED** [review]

File: index.js (hunk 3 above)
Comment (birme) [reply index.js]: You are correct. That line should be removed.

**Pull request merged**